classes/migration/upgrade/v3_4_0/I9813_QuickSubmitSubmissionProgressType.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I9813_QuickSubmitSubmissionProgressType.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I9813_QuickSubmitSubmissionProgressType
+ *
+ * @brief Fix the old submission_progress values inserted by QuickSubmit plugin,
+ *   from an int to a string to match the new step ids
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+use PKP\install\DowngradeNotSupportedException;
+
+class I9813_QuickSubmitSubmissionProgressType extends \PKP\migration\Migration
```

Comment:
@bozana, this migration is enough for the next release (3.5).
But for 3.4, I think it also needs to check if the user is using a broken `Quick Submit` plugin (version 1_0_7-1), and block the upgrade or it should at least emit an installation warning and/or a notification to the journal manager.

It's also important to release a new version of the `Quick Submit` with the latest fixes to stop the bleeding, but I don't have permissions to do that 😁

Reply:
I think we can change plugin.xml so that 1_0_7-1 version is only available until 3.4.0.5.
Then, with the next release >= 3.4.0.6 everyone should be without this bug, using this upgrade script.
Correct?

Yes, I have already spoken with Alec -- he would do it, I just wanted to double check if anything else would need to be considered for this next release... I will check the issues and ask in dev channel...

Reply:
I went through the open issues -- I do not think there is any needed to be considered immediately, so I will ask Alec to create a new release...

Reply:
Hmmm... I see...
If the user updates to 3.4.0.6 it could still have and continue to use QuickSubmit plugin 1_0_7-1.
Hmmm... OK...
I will then throw an error, so that the user first needs to update the QuickSubmit plugin.
OK?

Reply:
It would be nice to have this as a `pre-flight check`, but the current pre-flight check won't run again at minor upgrades, I'll create a discussion to discuss this 🤔

These magic values should be linked against the database, then we wouldn't have such issues and it would be easier to understand what are the possible values (I have no idea without inspecting the code): https://github.com/pkp/pkp-lib/discussions/7981.

Reply:
@jonasraoni, I have added the function that checks the plugin installed version and throws an exception.